test/e2e/installplan_e2e_test.go

```diff
@@ -731,4 +823,173 @@ func TestCreateInstallPlanWithPreExistingCRDOwners(t *testing.T) {
 		})
 		require.NoError(t, err)
 	})
+
+	t.Run("PreExistingCRDOwnerIsReplaced", func(t *testing.T) {
+		// Create the catalog source
+		_, cleanupCatalogSource, err := createInternalCatalogSource(t, c, crc, "mock-ocs-main", testNamespace, mainManifests, []extv1beta1.CustomResourceDefinition{dependentCRD, mainCRD}, []v1alpha1.ClusterServiceVersion{dependentBetaCSV, dependentStableCSV, mainStableCSV, mainBetaCSV})
+		require.NoError(t, err)
+		defer cleanupCatalogSource()
+
+		// Create a stable installplan
+		installPlanCleanup, err := decorateCommonAndCreateInstallPlan(t, c, crc, testNamespace, installPlan)
+		require.NoError(t, err)
+		defer installPlanCleanup()
+
+		// Wait for InstallPlan to be status: Complete before checking resource presence
+		fetchedInstallPlan, err := fetchInstallPlan(t, crc, installPlan.GetName(), installPlanCompleteOrFailedChecker)
+		require.NoError(t, err)
+		t.Logf("Install plan %s fetched with status %s", fetchedInstallPlan.GetName(), fetchedInstallPlan.Status.Phase)
+		require.True(t, installPlanCompleteChecker(fetchedInstallPlan))
+
+		// Esnure that the desired resources have been created
+		expectedSteps := map[registry.ResourceKey]struct{}{
+			registry.ResourceKey{Name: mainCRDName, Kind: "CustomResourceDefinition"}:                    {},
+			registry.ResourceKey{Name: dependentCRDName, Kind: "CustomResourceDefinition"}:               {},
+			registry.ResourceKey{Name: dependentPackageStable, Kind: v1alpha1.ClusterServiceVersionKind}: {},
+			registry.ResourceKey{Name: mainPackageStable, Kind: v1alpha1.ClusterServiceVersionKind}:      {},
+		}
+		require.Equal(t, len(expectedSteps), len(fetchedInstallPlan.Status.Plan))
+
+		for _, step := range fetchedInstallPlan.Status.Plan {
+			key := registry.ResourceKey{
+				Name: step.Resource.Name,
+				Kind: step.Resource.Kind,
+			}
+			_, ok := expectedSteps[key]
+			require.True(t, ok)
+
+			// Remove the entry from the expected steps set (to ensure no duplicates in resolved plan)
+			delete(expectedSteps, key)
+		}
+
+		// Should have removed every matching step
+		require.Equal(t, 0, len(expectedSteps))
```

Comment:
could you add an extra message to this? `require.Equal(t, expected, actual, msg, args...)`

just imagining hitting this error and seeing "expected steps should be zero" and finding that a little confusing